src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -261,27 +261,27 @@ class Tensor<X, Symm, IndexList<Indices...>> {
 
   // @{
   /// Retrieve a TensorExpression object with the index structure passed in
-  template <typename... N,
-            Requires<std::conjunction_v<tt::is_tensor_index<N>...> and
-                     tmpl::is_set<N...>::value> = nullptr>
-  SPECTRE_ALWAYS_INLINE constexpr TE<tmpl::list<N...>> operator()(
-      N... /*meta*/) const noexcept {
-    return TE<tmpl::list<N...>>(*this);
-  }
-
-  template <typename... N,
-            Requires<std::conjunction_v<tt::is_tensor_index<N>...> and
-                     not tmpl::is_set<N...>::value> = nullptr>
-  SPECTRE_ALWAYS_INLINE constexpr auto operator()(N... /*meta*/) const noexcept
-      -> decltype(
-          TensorExpressions::detail::fully_contract<
-              TE, replace_indices<tmpl::list<N...>, repeated<tmpl::list<N...>>>,
-              0, tmpl::size<repeated<tmpl::list<N...>>>>(*this)) {
-    using args_list = tmpl::list<N...>;
-    using repeated_args_list = repeated<args_list>;
-    return TensorExpressions::detail::fully_contract<
-        TE, replace_indices<args_list, repeated_args_list>, 0,
-        tmpl::size<repeated_args_list>>(*this);
+  template <
+      typename... TensorIndices,
+      Requires<std::conjunction_v<tt::is_tensor_index<TensorIndices>...>> =
```

Comment:
I think this `Requires` argument can be a `static_assert` it's mostly used for error message info, not for SFINAE and overload resolution.

Reply:
Done!